src/Tools/dotnet-monitor/CollectionRules/IAction.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules
+{
+    internal interface IAction<TOptions>
+    {
+        Task<ActionResponse> Execute(TOptions options, CancellationToken token);
```

Comment:
I think we're going to need to pass some notion of what the target process is to the action, especially for the collect actions. Maybe a `DiagnosticsClient client` parameter (or maybe some lower level notion that represents the connection to the target process).